llvmlite/binding/newpassmangers.py

```diff
@@ -0,0 +1,283 @@
+from ctypes import c_bool, c_int
+from llvmlite.binding import ffi
+
+
+def create_new_module_pass_manager():
+    return NewModulePassManager()
+
+
+def create_new_function_pass_manager():
+    return NewFunctionPassManger()
+
+
+def create_pass_builder(tm, pto):
+    return PassBuilder(tm, pto)
+
+
+def create_pipeline_tuning_options(opt_level=2, size_level=0):
+    return PipelineTuningOptions(opt_level=opt_level, size_level=size_level)
+
+
+class NewModulePassManager(ffi.ObjectRef):
+
+    def __init__(self, ptr=None):
+        if ptr is None:
+            ptr = ffi.lib.LLVMPY_CreateNewModulePassManager()
+        super().__init__(ptr)
+
```

Comment:
As far as I can tell there isn't a use case for passing in a pointer to an existing LLVM object, which I think is inherited from the style in `passmanagers.py`. Let's just get rid of that capability:

```suggestion
    def __init__(self):
        super().__init__(ffi.lib.LLVMPY_CreateNewModulePassManager())
```

Reply:
> As far as I can tell there isn't a use case for passing in a pointer to an existing LLVM object, which I think is inherited from the style in `passmanagers.py`. Let's just get rid of that capability:

Yes that was why I added them here, I can't think of a use for this feature from python user's pov

Reply:
But it's being used internally. Just checked `PassBuilder` relies on `ptr` based initialization of `MPM` and `FPM` objects, when we call `PB.getModulePassManager()` and `PB.getFunctionPassManager()` APIs respectively.

We can get rid of it at other 2 places though (PassBuilder and PipelineTuningOptions's constructors).

Reply:
Ah, OK - well spotted!